File: backoffice/Domain/Entidades/Cadastros/ContratoPrestacaoServico/ContratoPrestacaoServico.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text.Json.Serialization;

namespace Domain.Entidades.Cadastros.ContratoPrestacaoServico;

public class ContratoPrestacaoServico
{
    [Key]
    public int Id { get; set; }
    public string DistanciaPista { get; set; }
    public string Preco { get; set; }
    public string UnidadePreco { get; set; }
    public string Extensao { get; set; }
    public string ValorTotal { get; set; }
    public string Vencimento { get; set; }
    public string NomePiloto { get; set; }
    public string Executor { get; set; }
    
    [ForeignKey("Empresa")]
    public int? IdEmpresa { get; set; }

    [JsonIgnore]
    public virtual Empresa.Empresa? Empresa { get; set; }
}

File: app/flytec/lib/features/aplications/models/contrato_prestacao_servico.dart
class ContratoPrestacaoServico {
  String? distanciaPista;
  String? preco;
  String? unidadePreco;
  String? extensao;
  String? valorTotal;
  String? vencimento;
  String? nomePiloto;
  String? executor;
  int? id;

  ContratoPrestacaoServico({
    this.distanciaPista,
    this.preco,
    this.unidadePreco,
    this.extensao,
    this.valorTotal,
    this.vencimento,
    this.nomePiloto,
    this.executor,
    this.id,
  });

  Map<String, dynamic> toMap() {
    return {
      'distanciaPista': distanciaPista,
      'preco': preco,
      'unidadePreco': unidadePreco,
      'extensao': extensao,
      'valorTotal': valorTotal,
      'vencimento': vencimento,
      'nomePiloto': nomePiloto,
      'executor': executor,
    };
  }

  factory ContratoPrestacaoServico.fromJson(Map<String, dynamic>? json) {
    return ContratoPrestacaoServico(
      distanciaPista: json?['distanciaPista'] ?? '',
      preco: json?['preco'] ?? '',
      unidadePreco: json?['unidadePreco'] ?? '',
      extensao: json?['extensao'] ?? '',
      valorTotal: json?['valorTotal'] ?? '',
      vencimento: json?['vencimento'] ?? '',
      nomePiloto: json?['nomePiloto'] ?? '',
      executor: json?['executor'] ?? '',
      id: json?['id'] ?? 0,
    );
  }
}
